app/src/main/java/com/rob729/quiethours/Fragments/MainFragment.kt

```diff
@@ -108,6 +108,35 @@ class MainFragment : Fragment() {
         setHasOptionsMenu(true)
         return binding.root
     }
+    private fun deleteAll() {
+        if (binding.emrl.visibility == View.GONE) {
+            AlertDialog.Builder(requireContext())
+                .setTitle("Delete All Profiles")
+                .setMessage("Are you sure you want to delete all the profiles?")
+                .setPositiveButton("Yes") { _, dialogInterface ->
+                        val size: Int = profileListAdapter.itemCount
+                    if (size > 0) {
+                            for (i in 0 until size) {
+                                profileListAdapter.removeitem(i)
```

Comment:
Is this line needed?

Reply:
@rob729 Yes, without this the items are not being removed from the recycler view in MainFragment on selecting the delete all option

Reply:
IMO it will be better if you define the delete all function in the list adapter itself and call it in the alert dialog.

Reply:
Ok, if you say... I will try to do it in that manner.

Reply:
@rob729 Done